src/nupic/research/temporal_memory.py

```diff
@@ -406,19 +459,19 @@ def leastUsedCell(cells, connections, random):
 
 
   @staticmethod
-  def growSynapses(connections, initialPermanence, nDesiredNewSynapes,
-                   prevWinnerCells, random, segment):
+  def growSynapses(connections, random, segment, nDesiredNewSynapes,
+                   prevWinnerCells, initialPermanence):
```

Comment:
To match the C++. It puts the mutated arguments first, followed by traditional arguments for the function, followed by TM parameters.

This pattern is more intuitive in C++, where the types specify that the first few params aren't "const".